main.go

```diff
@@ -9,49 +9,61 @@ import (
 	"github.com/urfave/cli"
 )
 
-const (
-	cattleURLEnv       = "CATTLE_URL"
-	cattleAccessKeyEnv = "CATTLE_ACCESS_KEY"
-	cattleSecretKeyEnv = "CATTLE_SECRET_KEY"
-)
-
 var VERSION = "v0.0.0-dev"
 
 func main() {
 	app := cli.NewApp()
 	app.Name = "netes-agent"
 	app.Version = VERSION
-	app.Flags = []cli.Flag{}
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:   "access-key",
+			EnvVar: "CATTLE_ACCESS_KEY",
+			Usage:  "Rancher access key",
+		},
+		cli.StringFlag{
+			Name:   "secret-key",
+			EnvVar: "CATTLE_SECRET_KEY",
+			Usage:  "Rancher secret key",
+		},
+		cli.StringFlag{
+			Name:   "url",
+			Value: "http://localhost:8080/v3",
+			EnvVar: "CATTLE_URL",
+			Usage:  "Rancher URL",
+		},
+	}
 	app.Action = action
 	if err := app.Run(os.Args); err != nil {
 		log.Fatal(err)
 	}
 }
 
 func action(c *cli.Context) error {
-	cattleURL := os.Getenv(cattleURLEnv)
-	cattleAccessKey := os.Getenv(cattleAccessKeyEnv)
-	cattleSecretKey := os.Getenv(cattleSecretKeyEnv)
-
 	rancherClient, err := client.NewRancherClient(&client.ClientOpts{
-		Url:       cattleURL,
-		AccessKey: cattleAccessKey,
-		SecretKey: cattleSecretKey,
+		Url:       c.String("url"),
+		AccessKey: os.Getenv("access-key"),
```

Comment:
Don't you want `c.String("access-key")` here?

Reply:
Yep :) that's what I meant.